### PR TITLE
Changing boundary ID to boundary name mapping in orthogonal mesh generator

### LINKED ISSUE #229: Change boundary ID to name mapping in orthogonal meshes
Currently, the boundary ID to name mapping is:

1. 1D
    - `4` -> `ZMAX`
    - `5` -> `ZMIN`
3. 2D
    - `0` -> `XMAX`
    - `1` -> `XMIN`
    - `2` -> `YMAX`
    - `3` -> `YMIN`
4. 3D
    - `0` -> `XMAX`
    - `1` -> `XMIN`
    - `2` -> `YMAX`
    - `3` -> `YMIN`
    - `4` -> `ZMAX`
    - `5` -> `ZMIN`

Change it so it is:
1. 1D
    - `4` -> `ZMIN`
    - `5` -> `ZMAX`
3. 2D
    - `0` -> `XMIN`
    - `1` -> `XMAX`
    - `2` -> `YMIN`
    - `3` -> `YMAX`
4. 3D
    - `0` -> `XMIN`
    - `1` -> `XMAX`
    - `2` -> `YMIN`
    - `3` -> `YMAX`
    - `4` -> `ZMIN`
    - `5` -> `ZMAX`

### PR DESCRIPTION
**Pull request by andrsd**
Closes #229

### CODE REVIEW

File: framework/mesh/mesh_generator/orthogonal_mesh_generator.cc [hunk 1]
```diff
@@ -257,37 +254,33 @@ OrthogonalMeshGenerator::CreateUnpartitioned2DOrthoMesh(const std::vector<double
...
+          face.neighbor = cmap[i][j + 1]; /*XMIN*/
         if (v == 3 and j != 0)
-          face.neighbor = cmap[i][j - 1]; /*XMIN*/
+          face.neighbor = cmap[i][j - 1]; /*XMAX*/
         if (v == 2 and i != max_i)
-          face.neighbor = cmap[i + 1][j]; /*YMAX*/
+          face.neighbor = cmap[i + 1][j]; /*YMIN*/
         if (v == 0 and i != 0)
-          face.neighbor = cmap[i - 1][j]; /*YMIN*/
+          face.neighbor = cmap[i - 1][j]; /*YMAX*/
```
Comment (zhardy-lanl) [inline framework/mesh/mesh_generator/orthogonal_mesh_generator.cc:263]: These are interesting comments, aren't these interior cells here? I wonder why these comments were there in the first place... Either way, they should be reverted. The change in boundary ordering doesn't change the orientation of the cell during construction, especially not interior cells.

File: framework/mesh/mesh_generator/orthogonal_mesh_generator.cc [hunk 2]
```diff
@@ -257,37 +254,33 @@ OrthogonalMeshGenerator::CreateUnpartitioned2DOrthoMesh(const std::vector<double
...
 
         // boundary logic
         if (v == 1 and j == max_j)
         {
-          /*XMAX*/
-          face.neighbor = 0;
+          face.neighbor = XMIN;
```
Comment (zhardy-lanl) [inline framework/mesh/mesh_generator/orthogonal_mesh_generator.cc:268]: This doesn't seem right. The boundary is still `XMAX`. The thing that changed is the index of `XMAX`, not the boundary that this is.

File: framework/mesh/mesh_generator/orthogonal_mesh_generator.cc [hunk 3]
```diff
@@ -257,37 +254,33 @@ OrthogonalMeshGenerator::CreateUnpartitioned2DOrthoMesh(const std::vector<double
...
           face.has_neighbor = false;
         }
         if (v == 3 and j == 0)
         {
-          /*XMIN*/
-          face.neighbor = 1;
+          face.neighbor = XMAX;
```
Comment (zhardy-lanl) [inline framework/mesh/mesh_generator/orthogonal_mesh_generator.cc:273]: See other comment.

File: framework/mesh/mesh_generator/orthogonal_mesh_generator.cc [hunk 4]
```diff
@@ -257,37 +254,33 @@ OrthogonalMeshGenerator::CreateUnpartitioned2DOrthoMesh(const std::vector<double
...
           face.has_neighbor = false;
         }
         if (v == 2 and i == max_i)
         {
-          /*YMAX*/
-          face.neighbor = 2;
+          face.neighbor = YMIN;
```
Comment (zhardy-lanl) [inline framework/mesh/mesh_generator/orthogonal_mesh_generator.cc:278]: See other comment.

File: framework/mesh/mesh_generator/orthogonal_mesh_generator.cc [hunk 5]
```diff
@@ -257,37 +254,33 @@ OrthogonalMeshGenerator::CreateUnpartitioned2DOrthoMesh(const std::vector<double
...
           face.has_neighbor = false;
         }
         if (v == 0 and i == 0)
         {
-          /*YMIN*/
-          face.neighbor = 3;
+          face.neighbor = YMAX;
```
Comment (zhardy-lanl) [inline framework/mesh/mesh_generator/orthogonal_mesh_generator.cc:283]: See other comment.

File: framework/mesh/mesh_generator/orthogonal_mesh_generator.cc [hunk 6]
```diff
@@ -399,7 +392,7 @@ OrthogonalMeshGenerator::CreateUnpartitioned3DOrthoMesh(const std::vector<double
                                                   vmap[i + 1][j + 1][k],
                                                   vmap[i + 1][j + 1][k + 1],
                                                   vmap[i][j + 1][k + 1]};
-          face.neighbor = (j == max_j) ? 0 /*XMAX*/ : cmap[i][j + 1][k];
+          face.neighbor = (j == max_j) ? XMIN : cmap[i][j + 1][k];
```
Comment (zhardy-lanl) [inline framework/mesh/mesh_generator/orthogonal_mesh_generator.cc:395]: See other comment.

File: framework/mesh/mesh_generator/orthogonal_mesh_generator.cc [hunk 7]
```diff
@@ -409,7 +402,7 @@ OrthogonalMeshGenerator::CreateUnpartitioned3DOrthoMesh(const std::vector<double
 
           face.vertex_ids = std::vector<uint64_t>{
             vmap[i][j][k], vmap[i][j][k + 1], vmap[i + 1][j][k + 1], vmap[i + 1][j][k]};
-          face.neighbor = (j == 0) ? 1 /*XMIN*/ : cmap[i][j - 1][k];
+          face.neighbor = (j == 0) ? XMAX : cmap[i][j - 1][k];
```
Comment (zhardy-lanl) [inline framework/mesh/mesh_generator/orthogonal_mesh_generator.cc:405]: See other comment.

File: framework/mesh/mesh_generator/orthogonal_mesh_generator.cc [hunk 8]
```diff
@@ -421,7 +414,7 @@ OrthogonalMeshGenerator::CreateUnpartitioned3DOrthoMesh(const std::vector<double
                                                   vmap[i + 1][j][k + 1],
                                                   vmap[i + 1][j + 1][k + 1],
                                                   vmap[i + 1][j + 1][k]};
-          face.neighbor = (i == max_i) ? 2 /*YMAX*/ : cmap[i + 1][j][k];
+          face.neighbor = (i == max_i) ? YMIN : cmap[i + 1][j][k];
```
Comment (zhardy-lanl) [inline framework/mesh/mesh_generator/orthogonal_mesh_generator.cc:417]: See other comment.

File: framework/mesh/mesh_generator/orthogonal_mesh_generator.cc [hunk 9]
```diff
@@ -431,7 +424,7 @@ OrthogonalMeshGenerator::CreateUnpartitioned3DOrthoMesh(const std::vector<double
 
           face.vertex_ids = std::vector<uint64_t>{
             vmap[i][j][k], vmap[i][j + 1][k], vmap[i][j + 1][k + 1], vmap[i][j][k + 1]};
-          face.neighbor = (i == 0) ? 3 /*YMIN*/ : cmap[i - 1][j][k];
+          face.neighbor = (i == 0) ? YMAX : cmap[i - 1][j][k];
```
Comment (zhardy-lanl) [inline framework/mesh/mesh_generator/orthogonal_mesh_generator.cc:427]: See other comment.

File: framework/mesh/mesh_generator/orthogonal_mesh_generator.cc [hunk 10]
```diff
@@ -453,7 +446,7 @@ OrthogonalMeshGenerator::CreateUnpartitioned3DOrthoMesh(const std::vector<double
 
           face.vertex_ids = std::vector<uint64_t>{
             vmap[i][j][k], vmap[i + 1][j][k], vmap[i + 1][j + 1][k], vmap[i][j + 1][k]};
-          face.neighbor = (k == 0) ? 5 /*ZMIN*/ : cmap[i][j][k - 1];
+          face.neighbor = (k == 0) ? ZMAX : cmap[i][j][k - 1];
```
Comment (zhardy-lanl) [inline framework/mesh/mesh_generator/orthogonal_mesh_generator.cc:449]: See other comment.

**Review by zhardy-lanl — CHANGES_REQUESTED** [review]
If we are switching XMAX=0 -> XMIN=0, then the boundary IDs assigned to each boundary should change. It seems that the changes here simply calls XMAX XMIN when it is still XMAX in reality. For example, see [this](https://github.com/Open-Sn/opensn/pull/255/files#diff-eea216c0d703c277f363af126cfc51ad97540fdd1135309bac14922cd57036c8R266-R268).

It took me a few minutes to understand why the regressions tests are passing, and I think I know why now. In `LBSSolver`, the same ordering still exists, and since the indices did not change, there is no underlying difference. See [this](https://github.com/Open-Sn/opensn/blob/main/modules/linear_boltzmann_solvers/lbs_solver/lbs_solver.cc#L36).

**Review by zhardy-lanl — APPROVED** [review]

**Review by ragusa — APPROVED** [review]

**Review by wdhawkins — APPROVED** [review]

**Pull request merged**